audio/out/ao_wasapi.h

```diff
@@ -48,21 +48,22 @@ void wasapi_change_uninit(struct ao* ao);
 
 enum wasapi_thread_state {
     WASAPI_THREAD_FEED = 0,
-    WASAPI_THREAD_DISPATCH,
     WASAPI_THREAD_RESUME,
     WASAPI_THREAD_RESET,
-    WASAPI_THREAD_SHUTDOWN
+    WASAPI_THREAD_SHUTDOWN,
+    WASAPI_THREAD_DISPATCH_BASE
 };
 
 typedef struct wasapi_state {
     struct mp_log *log;
 
-    bool init_ok;            // status of init phase
+    bool init_ok;                  // status of init phase
     // Thread handles
-    HANDLE hInitDone;        // set when init is complete in audio thread
-    HANDLE hAudioThread;     // the audio thread itself
-    HANDLE hWake;            // thread wakeup event
-    atomic_int thread_state; // enum wasapi_thread_state (what to do on wakeup)
+    HANDLE hInitDone;              // set when init is complete in audio thread
+    HANDLE hAudioThread;           // the audio thread itself
+    HANDLE hWake;                  // thread wakeup event
+    _Atomic uint32_t thread_state; // enum wasapi_thread_state (what to do on wakeup)
+    _Atomic uint32_t dispatch_seq; // incremented each time AO control is called
```

Comment:
I would prefer `atomic_uint` for simplicity.

Reply:
Sure why not.

Reply:
Ah, I mistook it for `atomic_int`. Well that way we have even fewer changes. Can change it to `atomic_uint` if you'd like that though.

Reply:
Yeah, we need unsigned in case we overflow the (u)int_max and need to wrap around to 0. signed overflow is UB and even so we don't want to loop all the way back to zero in this case.

Reply:
> we don't want to loop all the way back to zero in this case.

```c
    do {
        seq = atomic_fetch_add(&state->dispatch_seq, 1);
    } while (seq >= 0 && seq < WASAPI_THREAD_DISPATCH_BASE);
```
This is already handled.

Reply:
Yeah, and would do like INT_MIN atomic additions after overflow.

Reply:
Changed to `atomic_uint` after all just in case.